coqlspclient/coq_structs.py

```diff
@@ -90,12 +90,41 @@ def __hash__(self) -> int:
         return hash(self.text)
 
 
-class ProofStep(Step):
-    def __init__(self, step: Step, goals: GoalAnswer, context: List[Term]):
-        super().__init__(step.text, step.ast)
-        self.goals = goals
+class ProofStep:
+    def __init__(
+        self,
+        step: Step,
+        goals: Union[GoalAnswer, Callable[[Position], GoalAnswer]],
+        context: List[Term],
+    ):
+        self.step = step
+        self._goals = goals
```

Comment:
Again, one underscore looks weird. Why not `__goals` here?

Reply:
The one underscore is because of what I said earlier, but I actually don't know why I used it here. I'll probably change it to two.